plugin/storage/es/spanstore/reader.go

```diff
@@ -209,20 +219,32 @@ func bucketToStringArray(buckets []*elastic.AggregationBucketKeyItem) ([]string,
 
 // FindTraces retrieves traces that match the traceQuery
 func (s *SpanReader) FindTraces(ctx context.Context, traceQuery *spanstore.TraceQueryParameters) ([]*model.Trace, error) {
+	span, ctx := startSpanForQuery(ctx, "FindTraces", traceQuery.ServiceName)
+	for k, v := range traceQuery.Tags {
+		span.LogFields(otlog.String("tag.key", k), otlog.String("tag.value", v))
+	}
+	defer span.Finish()
+
 	if err := validateQuery(traceQuery); err != nil {
 		return nil, err
 	}
 	if traceQuery.NumTraces == 0 {
 		traceQuery.NumTraces = defaultNumTraces
 	}
-	uniqueTraceIDs, err := s.findTraceIDs(traceQuery)
+	uniqueTraceIDs, err := s.findTraceIDs(ctx, traceQuery)
 	if err != nil {
 		return nil, err
 	}
-	return s.multiRead(uniqueTraceIDs, traceQuery.StartTimeMin, traceQuery.StartTimeMax)
+	return s.multiRead(ctx, uniqueTraceIDs, traceQuery.StartTimeMin, traceQuery.StartTimeMax)
 }
 
-func (s *SpanReader) multiRead(traceIDs []string, startTime, endTime time.Time) ([]*model.Trace, error) {
+func (s *SpanReader) multiRead(ctx context.Context, traceIDs []string, startTime, endTime time.Time) ([]*model.Trace, error) {
+
+	childSpan, _ := opentracing.StartSpanFromContext(ctx, "multiRead")
+	for k, v := range traceIDs {
+		childSpan.LogFields(otlog.String("tag.num", strconv.Itoa(k)), otlog.String("tag.value", v))
```

Comment:
why tag.num? I think you want a single log field `trace_ids` with value being the array of trace IDs.

Reply:
Right, will change this.